scripts/l2/l2-gen-addresses.sh

```diff
@@ -46,9 +80,15 @@ for role in ADMIN BATCHER PROPOSER; do
     address="${!address_var}"
     
     if [[ -n "$address" ]]; then
-        echo "Funding $role address: $address with amount $L1_FUND_AMOUNT"
-        fund_address "$address" "$L1_FUND_AMOUNT"
-        echo
+        balance=$(check_address_balance "$address")
+        if [[ "$balance" == "0" ]]; then
+            echo "Funding $role address: $address with amount $L1_FUND_AMOUNT"
+            fund_address "$address" "$L1_FUND_AMOUNT"
+            echo
+        else
+            echo "$role address: $address already funded"
+            echo
```

Comment:
when will this branch be hit? can we add a comment? is it expected case or should we exit 1?

Reply:
ah you're right, this path shouldnt trigger. updated to throw message and exit instead